### PR TITLE
feat: use modern query schema

### PR DESCRIPTION
**Pull request by ariser**
The new `schema` param in api call `viewer/json/query` switches between 3 response formats. This PR adds utility functions and types to request and handle these responses.

Also, a few bug fixes:
- In Preview, “Table is empty” is now only displays for tables, previously it displayed for folders
- The previous fix exposed a bug with handling error responses in Preview — in some cases it tried to render a non-`ReactNode` entity; fixed by always converting error to string
- adjusted Preview paddings to match the rest of the app

### CODE REVIEW

File: src/components/QueryResultTable/QueryResultTable.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    return (
+        <DataTable
+            columns={columns}
+            data={data}
+            settings={settings}
+            // to prevent accessing row.id in case it is present but is not the PK
+            // TODO ask someone before merge
+            rowKey={(_, index) => index}
```
Comment (ariser) [inline src/components/QueryResultTable/QueryResultTable.tsx:115]: to myself: remember to check this

**Comment by Raubzeug** [comment]
1) App crashes when click "Explain query".
2) Always Bad Request error when try to see preview.

**Review by Raubzeug — CHANGES_REQUESTED** [review]

**Comment by ariser** [comment]
> App crashes when click "Explain query".

Thank you for noticing! This is not caused by the changes in this PR, but it is connected to the corresponding backend changed. Will discuss this to make sure we preserve backward compatibility.

> Always Bad Request error when try to see preview.

This does not reproduce for me. Let's discuss this in PM if this is still an issue.

File: src/utils/query.ts [hunk 2]
```diff
@@ -7,6 +8,43 @@ export const isModern = (response: ClassicResponse | ModernResponse | YdbRespons
     Array.isArray(response.result) &&
     Array.isArray(response.result[0]);
 
+// eslint-disable-next-line complexity
+export const getCellType = (type: string) => {
+    switch (type.replace(/\?^/, '')) {
+        case YQLType.Bool:
+            return 'boolean';
```
[the remark this answers is not in the document]
Comment (ariser) [reply src/utils/query.ts:15]: I can add those, but I don't see an important reason why
I doubt type "number" will ever change to anything else :)
If you're convinced we should have them, can you explain in more detail?
Also, I renamed this function to `getColumnType` — if this changes anything for you

[the remark this answers is not in the document]
Comment (Raubzeug) [reply src/utils/query.ts:15]: Np, it is not critical at all. I just don't like string constants, but it's up to you of course!

**Review by Raubzeug — DISMISSED** [review]

**Review by Raubzeug — APPROVED** [review]

**Pull request merged**